src/net/connection/virtual_connection.rs

```diff
@@ -109,6 +109,17 @@ impl VirtualConnection {
 
         Ok(Some(Packet::new(self.remote_address, Box::from(payload), header.delivery_method)))
     }
+
+    /// This will gather dropped packets from the reliable channels.
+    ///
+    /// Note that after requesting dropped packets the dropped packets will be removed from this client.
+    pub fn gather_dropped_packets(&mut self) -> Option<Vec<Box<[u8]>>>{
```

Comment:
This could just be an empty Vec btw

Reply:
Reason why I do that is because I don't have to allocate a new vector if there are no dropped packets. I could simply return None, isn't this a better approach?

Reply:
Creating an empty `Vec` doesn't allocate anything.

Reply:
Nitpick: with_capacity is an option. =)

Reply:
And it does still take up a minuscule amount of bookkeeping memory

Reply:
@fhaynes should I go with Vec::with_capacity(0) or just Option. I have been at that decision and went with Option but watch your view?

Reply:
Talked with @TimonPost on Discord about this. Summary was to return `Vec::with_capacity(0)`.